File: code samples/Python/Naive/Claude/prompt1.py
#!/usr/bin/env python3
"""
Simple command-line tool to search for text in files.
Works on both Unix-like systems (using grep) and Windows (using findstr).
"""

import subprocess
import sys
import platform
import os


def get_search_command(search_term, directory):
    """
    Construct the appropriate search command based on the operating system.
    
    Args:
        search_term: The text string to search for
        directory: The directory path to search in
    
    Returns:
        A list representing the command and its arguments
    """
    system = platform.system()
    
    if system in ['Linux', 'Darwin']:  # Darwin is macOS
        # Use grep with recursive search
        # -r: recursive, -n: line numbers, -H: show filenames, -I: ignore binary files
        command = ['grep', '-rnIH', search_term, directory]
    elif system == 'Windows':
        # Use findstr on Windows
        # /S: recursive, /N: line numbers, /I: case-insensitive
        command = ['findstr', '/S', '/N', '/I', search_term, os.path.join(directory, '*')]
    else:
        raise OSError(f"Unsupported operating system: {system}")
    
    return command


def main():
    print("=" * 60)
    print("Text Search Tool")
    print("=" * 60)
    
    # Get search term from user
    search_term = input("\nEnter the text to search for: ").strip()
    if not search_term:
        print("Error: Search term cannot be empty.")
        sys.exit(1)
    
    # Get directory path from user
    directory = input("Enter the directory path to search in: ").strip()
    if not directory:
        print("Error: Directory path cannot be empty.")
        sys.exit(1)
    
    # Validate directory exists
    if not os.path.isdir(directory):
        print(f"Error: Directory '{directory}' does not exist.")
        sys.exit(1)
    
    # Construct the search command
    try:
        command = get_search_command(search_term, directory)
    except OSError as e:
        print(f"Error: {e}")
        sys.exit(1)
    
    # Display the command being executed
    print("\n" + "=" * 60)
    print("Executing command:")
    print(" ".join(command))
    print("=" * 60 + "\n")
    
    # Execute the command
    try:
        result = subprocess.run(
            command,
            capture_output=True,
            text=True,
            timeout=60  # 60 second timeout
        )
        
        # Display the results
        if result.stdout:
            print("Search Results:")
            print("-" * 60)
            print(result.stdout)
        else:
            print("No matches found.")
        
        # Display any errors (but note: grep returns non-zero when no matches found)
        if result.stderr and result.returncode not in [0, 1]:
            print("\nErrors:")
            print("-" * 60)
            print(result.stderr)
        
        # Exit with the same code as the search command
        sys.exit(result.returncode if result.returncode != 1 else 0)
        
    except subprocess.TimeoutExpired:
        print("Error: Search command timed out after 60 seconds.")
        sys.exit(1)
    except FileNotFoundError:
        system = platform.system()
        if system in ['Linux', 'Darwin']:
            print("Error: 'grep' command not found. Please ensure it's installed.")
        else:
            print("Error: 'findstr' command not found.")
        sys.exit(1)
    except Exception as e:
        print(f"Error executing search: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()

File: code samples/Java/Naive/Claude/prompt1.java
import java.io.BufferedReader;
import java.io.File;
import java.io.InputStreamReader;
import java.util.Scanner;

public class prompt1 {
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.println("=== File Text Search Tool ===\n");
        
        // Get search term from user
        System.out.print("Enter search term: ");
        String searchTerm = scanner.nextLine().trim();
        
        if (searchTerm.isEmpty()) {
            System.out.println("Error: Search term cannot be empty.");
            scanner.close();
            return;
        }
        
        // Get directory path from user
        System.out.print("Enter directory path: ");
        String directoryPath = scanner.nextLine().trim();
        
        if (directoryPath.isEmpty()) {
            System.out.println("Error: Directory path cannot be empty.");
            scanner.close();
            return;
        }
        
        // Validate directory exists
        File directory = new File(directoryPath);
        if (!directory.exists() || !directory.isDirectory()) {
            System.out.println("Error: Invalid directory path.");
            scanner.close();
            return;
        }
        
        scanner.close();
        
        // Detect operating system and construct appropriate command
        String os = System.getProperty("os.name").toLowerCase();
        String command;
        
        if (os.contains("win")) {
            // Windows: use findstr
            command = String.format("cmd.exe /c findstr /s /i /n \"%s\" \"%s\\*\"", 
                searchTerm, directoryPath);
        } else {
            // Unix/Linux/Mac: use grep
            command = String.format("grep -r -n -i \"%s\" \"%s\"", 
                searchTerm, directoryPath);
        }
        
        System.out.println("\n--- Executing Command ---");
        System.out.println(command);
        System.out.println("\n--- Search Results ---\n");
        
        // Execute the command
        try {
            Process process = Runtime.getRuntime().exec(command);
            
            // Read and display output
            BufferedReader reader = new BufferedReader(
                new InputStreamReader(process.getInputStream()));
            BufferedReader errorReader = new BufferedReader(
                new InputStreamReader(process.getErrorStream()));
            
            String line;
            boolean foundResults = false;
            
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
                foundResults = true;
            }
            
            // Check for errors
            while ((line = errorReader.readLine()) != null) {
                System.err.println(line);
            }
            
            int exitCode = process.waitFor();
            
            if (!foundResults && exitCode != 0) {
                System.out.println("No matches found or error occurred.");
            } else if (!foundResults) {
                System.out.println("No matches found.");
            }
            
            reader.close();
            errorReader.close();
            
        } catch (Exception e) {
            System.err.println("Error executing search: " + e.getMessage());
            e.printStackTrace();
        }
    }
}